src/server/pps/server/api_server.go

```diff
@@ -742,6 +742,21 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *ppsclient.Creat
 		return nil, err
 	}
 
+	existingPipelineInfo, err := persistClient.GetPipelineInfo(ctx, request.Pipeline)
```

Comment:
The motivation is we wanted the error to be more descriptive. Before the change if you try to create the same pipeline, you'll get an error like `repo foo exists`, which isn't descriptive. 

With the change, we check to see if the `PipelineInfo` matching the request already exists, and return a descriptive message (e.g. `pipeline xxx already exists`) if it does.

The race case is interesting. It won't race more than what happens today -- as you say the `CreatePipelineInfo` prevents that. But you're right - ideally we handle it here so that even in the race condition we return the nice error message.